### DISCUSSION TITLE
查询内存超限，sql被cancel之后BE内存和CPU不释放

### DISCUSSION THREAD

**Discussion by fdf1779** [apache/doris#23465] [A - General / Q&A]
在Doris中存储了一张每天1.3TB的数据，对其根据索引进行查询的时候会内存超限或者时间超限，但是SQL被CANCEL之后，BE所占用的CPU资源和内存资源依旧很高，好像之前被CANCEL的SQL查询时申请的资源并没有被释放一样，我该怎么避免这个问题。

**Comment by lin-xiaosheng** [comment]
蹲一下答案，我也出现类似问题

**Comment by LemonLiTree** [comment]
which doris version？

**Reply by fdf1779** [reply]
I have found my problem. It is not the memory occupied by the query, but in version 1.2. x, the data I inserted was too large to enable "enable_segcompaction=true", which resulted in a large number of merged files in the BE backend and uncontrollable memory

**Reply by LemonLiTree** [reply]
1.2 also supports Segment compaction, which should be after 1.2.2

**Reply by laotao** [reply]
> I have found my problem. It is not the memory occupied by the query, but in version 1.2. x, the data I inserted was too large to enable "enable_segcompaction=true", which resulted in a large number of merged files in the BE backend and uncontrollable memory

后来怎么解决的？

**Comment by nospeak12345** [comment]
我也蹲一下，我碰到一个问题，每次select完成后，内存就会少掉一点，不知怎么回事，似乎query_timeout到达时，又会释放回去一点。
